rdkit/Chem/SaltRemover.py

```diff
@@ -168,7 +170,7 @@ def _applyPattern(m, salt, notEverything):
     if dontRemoveEverything and len(Chem.GetMolFrags(mol)) <= 1:
       return mol
     modified = False
-    for i, salt in enumerate(self.salts):
+    for salt in self.salts:
       tMol = _applyPattern(mol, salt, dontRemoveEverything)
       if tMol is not mol:
```

Comment:
I know I wrote the original version of this, but this check doesn't make any sense.
I think you can just do:
```
for salt in self.salts:
  mol = _applyPattern(mol, salt, dontRemoveEverything)
```